web-app/django/VIM/apps/instruments/views/instrument_list.py

```diff
@@ -99,6 +102,30 @@ def get_queryset(self) -> QuerySet[Instrument]:
                 .select_related("thumbnail")
                 .prefetch_related(instrumentname_prefetch_manager)
             )
+
+        search_query = self.request.GET.get("q", "").strip()
+        solr_url = "http://solr:8983/solr/virtual-instrument-museum"
+
+        if search_query:
+            solr = pysolr.Solr(solr_url, timeout=10)
+            solr_params = {
+                "q": f"text:{search_query}",
+                "wt": "json",
+                "rows": 100,
+                "facet": "false",
+                "fl": "sid, instrument_names_s",
+            }
+
+            # Send query to Solr and retrieve results
+            solr_response = solr.search(**solr_params)
+
+            # Select instruments based on Solr results
+            instrument_ids = [result["sid"] for result in solr_response]
+            return (
```

Comment:
Why not just store the information you need to generate this list in Solr. Then you don't need to do a second database query for the results that you already know.